packages/ui/src/app/pages/Election/components/ElectionProgressBar.tsx

```diff
@@ -0,0 +1,623 @@
+        setDaysIndicator(defaultDaysIndicator)
+      }
+      setSelectedToolbarStage('')
+      return
+    }
+
+    if (props.electionStage === 'inactive') {
+      setStageDescription(selectedStage)
+      setVerbIndicator('begins in')
+
+      if (selectedStage === 'announcing') {
+        setDaysIndicator(defaultDaysIndicator)
+      } else if (selectedStage === 'voting') {
+        setDaysIndicator(convertDurationToTimeString(announcingEndBlock - currentBlock))
+      } else if (selectedStage === 'revealing') {
+        setDaysIndicator(convertDurationToTimeString(votingEndBlock - currentBlock))
+      }
+    } else if (props.electionStage === 'announcing') {
+      if (selectedStage === 'inactive') {
+        setStageDescription(selectedStage)
+        setVerbIndicator('ended on')
+        setDaysIndicator(inactiveEndDay)
+      } else {
+        setStageDescription(selectedStage)
+        setVerbIndicator('begins in')
+
+        if (selectedStage === 'voting') {
+          setDaysIndicator(defaultDaysIndicator)
+        } else if (selectedStage === 'revealing') {
+          setDaysIndicator(convertDurationToTimeString(votingEndBlock - currentBlock))
+        }
+      }
+    } else if (props.electionStage === 'voting') {
+      if (selectedStage === 'inactive') {
+        setStageDescription(selectedStage)
+        setVerbIndicator('ended on')
+        setDaysIndicator(inactiveEndDay)
+      } else if (selectedStage === 'announcing') {
+        setStageDescription(selectedStage)
+        setVerbIndicator('ended on')
+        setDaysIndicator(announcingEndDay)
+      } else if (selectedStage === 'revealing') {
+        setStageDescription(selectedStage)
+        setVerbIndicator('begins in')
+        setDaysIndicator(defaultDaysIndicator)
+      }
+    } else if (props.electionStage === 'revealing') {
+      setStageDescription(selectedStage)
+      setVerbIndicator('ended on')
+      if (selectedStage === 'announcing') {
+        setDaysIndicator(announcingEndDay)
+      } else if (selectedStage === 'voting') {
+        setDaysIndicator(votingEndDay)
+      } else if (selectedStage === 'inactive') {
+        setDaysIndicator(inactiveEndDay)
+      }
+    }
+    setSelectedToolbarStage(selectedStage)
+  }
+
+  return (
+    <MultiStatisticItem {...props}>
+      <StatisticItemSpacedContent key={duration}>
+        <StatisticBigLabel>
+          <StatisticBigLabel strong={true} style={{ textTransform: 'capitalize' }}>
+            {stageDescription}
+          </StatisticBigLabel>
+          <StatisticBigLabel strong={false}>{verbIndicator}</StatisticBigLabel>
+          <StatisticBigLabel>{daysIndicator}</StatisticBigLabel>
+        </StatisticBigLabel>
+        <StatisticLabel>{remainDays} days until next round</StatisticLabel>
+      </StatisticItemSpacedContent>
+
+      <ProgressBarLayout layout={progressBarAttr}>
+        <TooltipProgressBar
+          start={0}
+          end={inactiveProgress / 100}
+          color={inactiveColor}
```

Comment:
Please change this prop to a `isCurrent` boolean, then inside the `TooltipProgressBar`: `Colors.Blue[500]` is passed if the stage is current and `Colors.Blue[200]` if not

Reply:
Hello Thesan. Thanks for your detailed guideline and update. I will follow you and update this

Reply:
I have followed your suggestion except the refactoring it.  Please give me some more time to do this